### DISCUSSION TITLE
Customizing home page (landing/main page)

### DISCUSSION THREAD

**Discussion by evandde** [hbstack/.github#21] [Ideas]
I have a few suggestions about customization of the home page. 

Personally, I preferred the structure of hbs v1 (https://hbs.razonyang.com/v1/en/).

So here are some ideas:

1. Option to allow only featured posts at the top, with pinned posts in the post list.
2. Option to hide or move taxonomy icons (between feature/pinned posts and the post list)
3. Enable one or two-column display in the post list.
4. Sidebar for taxonomy icons or author profiles.

Thank you.

![image](https://github.com/hbstack/.github/assets/57508537/29ca02ea-a3dd-475d-a226-a75aa92467b9)

**Comment by razonyang** [comment]
I don't have solution for this so far, every one has his own taste, it's hard to satisfy all users.
The ideal way to do that is modularizing everything, so that users can pick up the layout they want, but it's too complex, and lack of solutions at this moment.

**Reply by evandde** [reply]
If I want to customize by myself, which module do i need to modify? 

could you let me know about that?

**Reply by razonyang** [reply]
The [blog](https://github.com/hbstack/blog) module. For the home layout, you need to modify the https://github.com/hbstack/blog/blob/main/layouts/partials/hb/modules/blog/index.html by creating your own to override it.

```
echo "MY HOMEPAGE" > layouts/partials/hb/modules/blog/index.html
```

**Reply by evandde** [reply]
Oh.. I thought that the blog module handles the page of each post.

It is quite confusing for me... 

If I change the index.html, will it affect to the post page? (I don't know the exact term to call it sorry)

**Reply by razonyang** [reply]
Only the homepage will be affected, that partial was included by the [layouts/_default/index.html](https://github.com/hbstack/blog/blob/main/layouts/_default/index.html). See also

- https://gohugo.io/templates/lookup-order/

**Reply by evandde** [reply]
Thanks to your module guide (https://hbstack.dev/en/develop/module/) now I can modify and test the homepage.

I have another concern.
After I change the "layouts/partials/hb/modules/blog/index.html" in my own blog module,
how can I follow the update of your original [blog](https://github.com/hbstack/blog) module for the other files?

Currently I am using local module in "layouts/modules/blog" with `replace github.com/hbstack/blog => ./layouts/modules/blog` in go.mod.

**Reply by razonyang** [reply]
You should create the `layouts/partials/hb/modules/blog/index.html` on your site instead of using a local modified module, it won't work when deployment, since the local module changes only work on your local machine.

> how can I follow the update of your original [blog](https://github.com/hbstack/blog) module for the other files?

It's no guarantee that the file won't change, you've to watch and adjust the changes of custom files.

**Comment by razonyang** [comment]
Some news, now the blog module/layout support an optional sidebar, preview it on [https://theme.hbstack.dev/en/](https://theme.hbstack.dev/en/) and checkout the [docs](https://hbstack.dev/en/modules/blog/sidebar/overview/) for setting it up.

**Comment by luigi8bits** [comment]
@razonyang I would like to change my home page layout from "blog" to "docs" layout, it's possible? 

I want to use only docs "theme" on my site, I want to create a "second brain" and docs layout is so perfect for that, can you help me?

**Reply by razonyang** [reply]
For customizing the homepage, you can use the `landing` layout by setting the `layout` as `landing` on front matter, for example, https://github.com/hbstack/site/blob/main/content/_index.md (preview: https://hbstack.dev/).

```yaml
---
title: My Home Page
layout: landing
---

HTML BODY CONTENT HERE.
```

For other list and single pages that not belong to `content/docs` folder, you can specify the `type` as `docs` on front matter, then the pages will be rendered in docs layout.

```yaml
---
type: docs
---
```

**Comment by razonyang** [comment]
With the new [cards style theme](https://hbstack.dev/themes/cards/) was out, [blog's `v0.23.0` and `v0.24.0` were released](https://hbstack.dev/2023/12/whats-news-in-blogs-v0.23.0-v0.24.0-and-related-modules/), we can easilly manage to customize the homepage and blog's layout. So I'm closing this discussion now.
